src/types/channel_categories.ts

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {$ID, IDMappedObjects, RelationOneToOne} from './utilities';
+import {Team} from './teams';
+
+export type ChannelCategoryType = 'favorites' | 'public' | 'private' | 'direct_messages' | 'custom';
+
+export type ChannelCategory = {
+    id: string;
+    team_id: $ID<Team>;
+    type: ChannelCategoryType;
+    display_name: string;
+
+    // channel_ids: $ID<Channel>;
```

Comment:
isn't needed anymore?

Reply:
This will be added in phase 2 once the server supplies the IDs. I'll add a couple comments